airsonic-main/src/main/java/org/airsonic/player/service/search/SearchServiceUtilities.java

```diff
@@ -1,200 +1,176 @@
-    };
-
-    public final BiConsumer<List<MediaFile>, Integer> addMediaFileIfAnyMatch = (dist, id) -> {
-        if (!dist.stream().anyMatch(m -> id == m.getId())) {
-            MediaFile mediaFile = mediaFileService.getMediaFile(id);
-            if (!isEmpty(mediaFile)) {
-                dist.add(mediaFile);
-            }
-        }
-    };
-
-    public final BiConsumer<List<Artist>, Integer> addArtistId3IfAnyMatch = (dist, id) -> {
-        if (!dist.stream().anyMatch(a -> id == a.getId())) {
-            Artist artist = artistDao.getArtist(id);
-            if (!isEmpty(artist)) {
-                dist.add(artist);
-            }
-        }
-    };
-
-    public final Function<Class<?>, @Nullable IndexType> getIndexType = (assignableClass) -> {
-        IndexType indexType = null;
-        if (assignableClass.isAssignableFrom(Album.class)) {
-            indexType = IndexType.ALBUM_ID3;
-        } else if (assignableClass.isAssignableFrom(Artist.class)) {
-            indexType = IndexType.ARTIST_ID3;
-        } else if (assignableClass.isAssignableFrom(MediaFile.class)) {
-            indexType = IndexType.SONG;
-        }
-        return indexType;
-    };
-
-    public final Function<Class<?>, @Nullable String> getFieldName = (assignableClass) -> {
-        String fieldName = null;
-        if (assignableClass.isAssignableFrom(Album.class)) {
-            fieldName = FieldNames.ALBUM;
-        } else if (assignableClass.isAssignableFrom(Artist.class)) {
-            fieldName = FieldNames.ARTIST;
-        } else if (assignableClass.isAssignableFrom(MediaFile.class)) {
-            fieldName = FieldNames.TITLE;
-        }
-        return fieldName;
-    };
-
-    public final BiConsumer<List<Album>, Integer> addAlbumId3IfAnyMatch = (dist, subjectId) -> {
-        if (!dist.stream().anyMatch(a -> subjectId == a.getId())) {
-            Album album = albumDao.getAlbum(subjectId);
-            if (!isEmpty(album)) {
-                dist.add(album);
-            }
-        }
-    };
-
-    private final Function<String, File> getRootDirectory = (version) -> {
-        return new File(SettingsService.getAirsonicHome(), version);
-    };
-
-    public final BiFunction<String, IndexType, File> getDirectory = (version, indexType) -> {
-        return new File(getRootDirectory.apply(version), indexType.toString().toLowerCase());
-    };
-
-    public final Term createPrimarykey(Album album) {
-        return new Term(FieldNames.ID, Integer.toString(album.getId()));
-    };
-
-    public final Term createPrimarykey(Artist artist) {
-        return new Term(FieldNames.ID, Integer.toString(artist.getId()));
-    };
-
-    public final Term createPrimarykey(MediaFile mediaFile) {
-        return new Term(FieldNames.ID, Integer.toString(mediaFile.getId()));
-    };
-
-    public final boolean addIgnoreNull(Collection<?> collection, Object object) {
-        return CollectionUtils.addIgnoreNull(collection, object);
-    }

```

Comment:
This is somewhat weird code by design.

It does the same (compare with equals) as with legacy, but does not have duplicate input values ​​when this code is called.
It suggests that there was some anxiety at the time of implementation, there were uninvestigated problems, or there was a simple misunderstanding.

- If we want to compare strictly, we should compare id.
- Even if you compare ids, this code will not work on the data flow.

If this code is removed and duplicate results are displayed, it indicates that the defect exists prior to the processing of the search.
There is no problem even if it is left as it is, but it is ultimately unnecessary code.